### ISSUE TITLE
Connection refused Exception after multiple successfull requests

### ISSUE THREAD

**Issue by escamoteur** [dart-lang/sdk#38227]
Hi,

I'm working on a Flutter App with a custom map view. Map parts are downloaded from a server Application that runs on my local machine.
My App on the **Emulator**  accesses the local machine over its external IP-Address

After serveral sucessfull requests it can be up to 300 I get a socket refused connection exception with a completely different PORT that the one I selected for the request see screenshot in debug mode.

![image](https://user-images.githubusercontent.com/16635729/64347508-5a920680-cff4-11e9-9b5d-ab2f599b0441.png)

It seems this only happens on the emulator and not on a real device.

````
Flutter 1.9.8-pre.83 • channel master • https://github.com/flutter/flutter
Framework • revision 3932ffb1cd (11 hours ago) • 2019-09-04 23:13:16 -0400
Engine • revision e7f9ef6aa0
Tools • Dart 2.5.0 (build 2.5.0-dev.4.0 36985859e4)```
```

**Comment by natebosch** [comment]
> I get a socket refused connection exception with a completely different PORT that the one I selected for the request

This is likely an example of https://github.com/dart-lang/sdk/issues/12693

My guess is this is most likely something to do with the emulator or the server, and the confusing part is that the exception shows a strange port number. You might as for help from the flutter team about whether this could be related to the emulator. https://github.com/flutter/flutter/issues/29688

**Comment by escamoteur** [comment]
Yep, but it seems to be the client port and not the server port as  described in #12693.
The strange thing is that it work for quite some time and the Exception doesn't tell me any more than this.

**If I configure the HttpClient to use a proxy (I was trying to catch the bastard with Fiddler) this error doesn't happen. May @Hixie has an idea.
